### ISSUE TITLE
IndentationError: unexpected indent

### ISSUE THREAD

**Issue by fcappucci** [plasma-umass/scalene#671]
**Describe the bug**
I am trying to use scalene to improve a rather big program code, developed to re-grid satellite's data observations.
The code is running w/o error within the python environment (Python 3.11.4), the libraries ingested by the code are numpy, scipy, netCDF4, os, shutil. No parallel computation is implemented at current stage.

When I run the code via scalene I've got the following error message:

Scalene: An exception of type IndentationError occurred. Arguments:
('unexpected indent', ('<unknown>', 50, 4, '    Main class\n', 50, -1))
Traceback (most recent call last):
  File "/nethome/cappufa/miniconda3/envs/scalenv/lib/python3.11/site-packages/scalene/scalene_profiler.py", line 2118, in run_profiler
    exit_status = profiler.profile_code(
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "/nethome/cappufa/miniconda3/envs/scalenv/lib/python3.11/site-packages/scalene/scalene_profiler.py", line 1904, in profile_code
    did_output = Scalene.output_profile()
                 ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nethome/cappufa/miniconda3/envs/scalenv/lib/python3.11/site-packages/scalene/scalene_profiler.py", line 945, in output_profile
    json_output = Scalene.__json.output_profiles(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nethome/cappufa/miniconda3/envs/scalenv/lib/python3.11/site-packages/scalene/scalene_json.py", line 415, in output_profiles
    enclosing_regions = ScaleneAnalysis.find_regions(code_str)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nethome/cappufa/miniconda3/envs/scalenv/lib/python3.11/site-packages/scalene/scalene_analysis.py", line 106, in find_regions
    tree = ast.parse(src)
           ^^^^^^^^^^^^^^
  File "/nethome/cappufa/miniconda3/envs/scalenv/lib/python3.11/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<unknown>", line 50
    Main class
IndentationError: unexpected indent

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]  Linux Centos 7.9
 - Version [e.g. 22] Scalene version 1.5.26 (2023.08.22) installed via conda install -c conda-forge scalene

**Comment by emeryberger** [comment]
Never seen this before! If you can possibly share the code, that would help, though a minimum working example would be even better.

**Comment by fcappucci** [comment]
Thank you for your prompt reply.
After my fist attempt, I initialized a new python environment specifically designed for scalene, strictly following the requirements.txt specifications, in previous test I was using python 3.11 (maybe the guilty?). After few tries the problem disappeared and I managed to have the profile.html result. 
I'll keep you posted in case the problem occurs again, and/or, if you'd like to try to reproduce the error, I can provide a full example of my code (which however requires also a minimum set of input data, total weight ~200MB).
